src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -612,10 +602,8 @@ private void RestoreEvents_SolutionRestoreCompleted(SolutionRestoredEventArgs ar
 
         private bool ShouldNoOpDueToRestore(SolutionRestoredEventArgs latestRestore)
         {
-            return latestRestore != null &&
-                   _currentRestore != null &&
-                   latestRestore.SolutionSpecHash == _currentRestore.SolutionSpecHash &&
-                   latestRestore.IsSuccess == _currentRestore.IsSuccess;
+            return _currentRestore != null &&
+                   latestRestore?.RestoreStatus == NuGetOperationStatus.NoOp;
```

Comment:
This'd change/regress the behavior. Compare with previous version: it executed scripts regardless of `IsSuccess` status.